kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,173 @@
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriterS3`` saves matplotlib objects as image files to s3.
+"""
+
+import copy
+import io
+from typing import Any, Dict, List, Optional, Union
+
+import boto3
+from matplotlib.pyplot import figure
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriterS3(AbstractDataSet):
+    # pylint: disable=too-many-instance-attributes
+    """``MatplotlibWriter`` saves matplotlib objects as image files.
+
+        Example:
+        ::
+
+            import matplotlib.pyplot as plt
+            from kedro.contrib.io.matplotlib_s3_writer import MatplotlibWriterS3
+
+            plt.plot([1, 2, 3], [4, 5, 6])
+
+            # Saving single plot
+            single_plot_writer = MatplotlibWriterS3(
+                bucket="my-super-great-bucket", filepath="matplot_lib_single_plot.png"
+            )
+            single_plot_writer.save(plt)
+
+            # Saving dictionary of plots (with SSE)
+            plots_dict = {}
+            for colour in ["blue", "green", "red"]:
+                plots_dict[colour] = plt.figure()
+                plt.plot([1, 2, 3], [4, 5, 6], color=colour)
+                plt.close()
+
+            dict_plot_writer = MatplotlibWriterS3(
+                bucket="my-super-great-bucket",
+                s3_put_object_args={"ServerSideEncryption": "AES256"},
+                filepath="matplotlib_dict",
+            )
+            dict_plot_writer.save(plots_dict)
+
+            # Saving list of plots
+            plots_list = []
+            for index in range(5):
+                plots_list.append(plt.figure())
+                plt.plot([1,2,3],[4,5,6], color=colour)
+            list_plot_writer = MatplotlibWriterS3(
+                bucket="my-super-great-bucket", filepath="matplotlib_list"
+            )
+            list_plot_writer.save(plots_list)
+
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        bucket: str,
```

Comment:
```suggestion
        bucket_name: str,
```
for consistency with our other S3 datasets :)

Reply:
Agree on consistency